File: MinerApplication/MinerApplication/BankingGold.cpp
#include "stdafx.h"
#include "BankingGold.h"
#include "MiningForGold.h"
BankingGold::BankingGold()
{
}

BankingGold::~BankingGold()
{
}

void BankingGold::Execute(Miner* miner)
{
	//Print out information of gold
	cout << "Banking Gold" << endl;
	//Banking the Miners gold, this adds the players current gold and then sets it to 0
	miner->m_BankedGold+= miner->m_Gold;
	miner->m_Gold = 0;
}

File: Miner Application(Unity)/Assets/Scripts/Miner/BankingGold.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
[System.Serializable]
public class BankingGold : State<Miner>
{
    public BankingGold()
    {

    }
    public override void Execute(Miner agent)
    {
        agent.TargetLocation = new Vector3(0, 1, 0);
        //Print out information of gold
        Debug.Log("Banking Gold");
        //Banking the Miners gold
        agent.MinerTimer += Time.deltaTime;
        if (agent.MinerTimer >= 30)
        {
            agent.m_BankedGold += agent.m_Gold;

            agent.m_Gold = 0;


            agent.m_CanShop = true;
        }
    }
}
